File: python/linear_search.py
import random


def busqueda(lista, objetivo):
    """Search for a number in a random list of integers.
    Big-O Complexity: O(n)"""
    match = False

    for elemento in lista:
        if elemento == objetivo:
            match = True
            break

    return match


list_int = [random.randint(0, 100) for i in range(20)]
ENCONTRADO = busqueda(list_int, 7)
print(list_int)
print("esta" if ENCONTRADO else "no esta")

File: lua/linear_search.lua
local ranlist = require "random_list"

function busqueda(lista, objetivo)
-- Search for a number in a random list of integers. Big-O Complexity: O(n)
  match = false

  for _, elemento in ipairs(lista) do
    if elemento == objetivo then
      match = true
      break
    end
  end

  return match
end

list_int = ranlist.ints(20)
ENCONTRADO = busqueda(list_int, 7)
for _, elemento in ipairs(list_int) do io.write(elemento, " ") end
if ENCONTRADO then print("\nel 7 esta") else print("\nel 7 no esta") end
